src/runtime-rs/crates/hypervisor/src/qemu/inner.rs

```diff
@@ -513,9 +512,15 @@ use crate::device::DeviceType;
 
 // device manager part of Hypervisor
 impl QemuInner {
-    pub(crate) async fn add_device(&mut self, device: DeviceType) -> Result<DeviceType> {
+    pub(crate) async fn add_device(&mut self, mut device: DeviceType) -> Result<DeviceType> {
         info!(sl!(), "QemuInner::add_device() {}", device);
-        self.devices.push(device.clone());
+        if self.qmp.is_some() {
```

Comment:
From functionality I think, it works well. Actually, It expresses the device inserted into guest with cold-plug(VM non-running) or hot-plug(VM running). I know this is also a good way(qmp Some/None) to do so. But just using `self.qmp.is_some()` will make new contributors confused, IMHO, comments added here might be better.

Reply:
Good point, I'll make that clearer.